qiskit/transpiler/passes/optimization/template_matching/forward_match.py

```diff
@@ -79,49 +79,49 @@ def __init__(
         # Transformation of the carg indices of the circuit to be adapted to the template indices
         self.carg_indices = []
 
+        self.successorstovisit = {}
+        self.isblocked = {}
+        self.matchedwith = {}
+
```

Comment:
Great! So these are now used instead of the information that used to be stored on DAGDepNodes. This is much much cleaner.

Reply:
The only leftover that is not purely in the template code is `qindices` and `cindices` which are computed and stored at the DAGDependencyV2 level. It's done in the `add_op_node` method. It's just a list of indices for the qargs and cargs for a node. It seemed easiest to store this at this level, rather than having to rebuild the list each time it's needed for a node in the template code. This can probably be optimized with a template code rework.